app/gist/controller.js

```diff
@@ -74,6 +75,11 @@ export default Em.Controller.extend({
   }),
 
   rebuildApp: Em.observer('model.files.@each.content', 'isAutorun', function() {
+    if (!this.get('unsaved')) {
+      Em.run.scheduleOnce('sync', function() {
```

Comment:
`Ember.run.scheduleOnce` does not really work with anonymous functions (because it uses `func1 === func2` to know if it should schedule again, and in the anon function case they are always different).  Can probably do `Ember.run(() => this.set('unsaved', true));` instead.